pandas/tests/scalar/timestamp/test_timestamp.py

```diff
@@ -1092,3 +1093,23 @@ def test_constructor_ambigous_dst():
     expected = ts.value
     result = Timestamp(ts).value
     assert result == expected
+
+
+@pytest.mark.xfail(
+    LooseVersion(compat._optional._get_version(dateutil)) < LooseVersion("2.7.0"),
+    reason="dateutil moved to Timedelta.total_seconds() in 2.7.0",
+)
+@pytest.mark.parametrize("epoch", [1552211999999999872, 1552211999999999999])
+def test_constructor_before_dst_switch(epoch):
+    # GH 31043
+    # Make sure that calling Timestamp constructor
+    # on time just before DST switch doesn't lead to
+    # nonexistent time or value change
+    # Works only with dateutil >= 2.7.0 as dateutil overrid
+    # pandas.Timedelta.total_seconds with
+    # datetime.timedelta.total_seconds before
```

Comment:
This does not make sense to me - `datetime.timedelta.total_seconds` should succeed, because it's equivalent to:

```python
def total_seconds(td):
  useconds = td.days * 86400
  useconds += td.seconds 
  useconds *= 1000000
  useconds += td.microseconds
  return useconds / 1e6
```

I think there's actually a deeper issue here, which is that `td.microseconds` and `td.seconds` are *rounded* rather than *truncated*. Consider this:

```python
def to_ns(td):
  ns = td.days * 86400
  ns += td.seconds
  ns *= 1000000
  ns += td.microseconds
  ns *= 1000
  ns += td.nanoseconds
  return ns

td = timedelta(1552211999999999872, unit="ns")
print(td.value)  # 1552211999999999872
print(to_ns(td))  # 1552212000000000872
```

That seems to be the actual root cause of this issue and should probably be fixed.